java/ql/src/experimental/Security/CWE/CWE-117/LogInjection.qll

```diff
@@ -0,0 +1,84 @@
+/**
+ * Provides a taint-tracking configuration for reasoning about untrusted user input used in log entries.
+ */
+
+import java
+import semmle.code.java.dataflow.FlowSources
+
+module LogInjection {
+  string logLevel() {
+    result = "trace" or
+    result = "info" or
+    result = "warn" or
+    result = "error" or
+    result = "fatal"
+  }
+
+  /**
+   * A data flow source for user input used in log entries.
+   */
+  abstract class Source extends DataFlow::Node { }
+
+  /**
+   * A data flow sink for user input used in log entries.
+   */
+  abstract class Sink extends DataFlow::Node { }
+
+  /**
+   * A sanitizer for malicious user input used in log entries.
+   */
+  abstract class Sanitizer extends DataFlow::Node { }
+
+  /**
+   * A taint-tracking configuration for untrusted user input used in log entries.
+   */
+  class LogInjectionConfiguration extends TaintTracking::Configuration {
+    LogInjectionConfiguration() { this = "LogInjection" }
+
+    override predicate isSource(DataFlow::Node source) { source instanceof Source }
+
+    override predicate isSink(DataFlow::Node sink) { sink instanceof Sink }
+
+    override predicate isSanitizer(DataFlow::Node node) { node instanceof Sanitizer }
+  }
+
+  /**
+   * A source of remote user controlled input.
+   */
+  class RemoteSource extends Source {
+    RemoteSource() { this instanceof RemoteFlowSource }
+  }
+
+  /**
+   * A class of popular logging utilities.
+   */
+  class LoggingType extends RefType {
```

Comment:
Instead of defining this, suggest factoring out https://github.com/github/codeql/blob/main/java/ql/src/experimental/Security/CWE/CWE-532/SensitiveInfoLog.ql#L31 as well as the logging methods used there (though we may not need the restriction to only low-priority messages seen in that query)

I'd say it belongs at `java/ql/src/semmle/code/java/dataflow/FlowSinks.qll`, alongside the existing `FlowSources.qll`.